misc/metrics/src/protocol_stack.rs

```diff
@@ -0,0 +1,13 @@
+use libp2p_core::multiaddr::{Multiaddr,Protocol};
+
+//TODO: remove this file and replace with calls to Multiaddr::protocol_stack
+//  once that lands : https://github.com/multiformats/rust-multiaddr/pull/60
+//  This does imply changing the keys from Vec<String> to Vec<'static str>
```

Comment:
Not quite. The idea I wanted to convey is that from a user PoV, an extension trait makes the interface look like it is already on `Multiaddr`, i.e. we are _extending_ `Multiaddr` with functionality that we would like to have there. Once you do `impl MultiaddrExt for Multiaddr`, you can just call `multiaddr.protocol_stack()` as if `Multiaddr` would natively provide that function.

That way, call-sites don't have to change once we do actually upstream it. At that point, the trait can be deleted and the function just goes onto a regular `impl Multiaddr` block.

Semantically, it doesn't make a difference but I find that an extension trait conveys a different intent compared to a module-private utility function :)

I don't feel strongly about it though so feel free to go with whatever!

Reply:
Don't mind the above, I didn't realize you were already creating the function upstream. All of this was so that we don't _need_ to create it upstream but can instead iterate on it here first.